frontend/javascripts/oxalis/view/action-bar/tracing_actions_view.js

```diff
@@ -241,9 +262,35 @@ class TracingActionsView extends React.PureComponent<Props, State> {
   };
 
   handleFinishAndGetNextTask = async () => {
+    UserLocalStorage.setItem(
+      "lastFinishedTask",
+      JSON.stringify({ annotationId: this.props.annotationId, finishedTime: Date.now() }),
+    );
     api.tracing.finishAndGetNextTask();
   };
 
+  handleReopenTask = async () => {
+    const localStorageEntry = UserLocalStorage.getItem("lastFinishedTask");
+    if (!localStorageEntry) return;
+
+    const { annotationId } = JSON.parse(localStorageEntry);
+
+    if (annotationId) {
+      await Model.ensureSavedState();
```

Comment:
You can move this line into the `onOk` function. That way, the UI reacts faster and if the user cancels the modal, the total waiting time was shorter.